src/auth/auth.ts

```diff
@@ -200,6 +200,36 @@ export class BaseAuth<T extends AbstractAuthRequestHandler> {
       });
   }
 
+  /**
+   * Gets the user data for the user corresponding to a given provider id.
+   *
+   * See [Retrieve user data](/docs/auth/admin/manage-users#retrieve_user_data)
+   * for code samples and detailed documentation.
+   *
+   * @param providerId The provider ID, for example, "google.com" for the
+   *   Google provider.
+   * @param providerUid The user identifier for the given provider.
+   *
+   * @return A promise fulfilled with the user data corresponding to the
+   *   given provider id.
+   */
+  public getUserByProviderId(providerId: string, providerUid: string): Promise<UserRecord> {
+    // Although we don't really advertise it, we want to also handle
+    // non-federated idps with this call. So if we detect one of them, we'll
+    // reroute this request appropriately.
+    if (providerId === 'phone') {
+      return this.getUserByPhoneNumber(providerUid);
+    } else if (providerId === 'email') {
```

Comment:
Could authRequestHandler.getAccountByFederatedId be renamed to getAccountByProviderId (or Uid?) as well?
The logic for 'phone' and 'email' would need to go into authRequestHandler in that case.

Reply:
It could. But I've been interpreting the auth-api-request.ts file as the object that implements the backend API. And since the backend makes a distinction between federated ids and non-federated ids, I thought that worth preserving here. (NB: my interpretation may be incorrect.)

Though it probably should at least use Uid rather than Id. (Fixed).

Reply:
My personal preference would be to avoid the federated[Id/Uid] terminology where possible, so that it doesn't get confused with the more general provider[Id/Uid] term.

In the long run, we should probably move towards renaming the field in the backend api as well. So, it would be reasonable to move the logic to auth-api-request.ts even if the file represents the backend implementation API.

Having said that, I would be OK with keeping the logic in auth.